encoding/encoding.go

```diff
@@ -1,36 +1,30 @@
 package encoding
 
-import (
-	"bytes"
-	"encoding/gob"
-
-	"github.com/Layr-Labs/eigenda/pkg/kzg/bn254"
-)
-
-// Proof is the multireveal proof
-// Coeffs is identical to input data converted into Fr element
-type Frame struct {
-	Proof  bn254.G1Point
-	Coeffs []bn254.Fr
+type Decoder interface {
+	// Decode takes in the chunks, indices, and encoding parameters and returns the decoded blob
+	Decode(chunks []*Frame, indices []ChunkNumber, params EncodingParams, inputSize uint64) ([]byte, error)
 }
 
-func (f *Frame) Encode() ([]byte, error) {
-	var buf bytes.Buffer
-	enc := gob.NewEncoder(&buf)
-	err := enc.Encode(f)
-	if err != nil {
-		return nil, err
-	}
-	return buf.Bytes(), nil
+type Prover interface {
+	Decoder
+	// Encode takes in a blob and returns the commitments and encoded chunks. The encoding will satisfy the property that
+	// for any number M such that M*params.ChunkLength > BlobCommitments.Length, then any set of M chunks will be sufficient to
+	// reconstruct the blob.
+	EncodeAndProve(data []byte, params EncodingParams) (BlobCommitments, []*Frame, error)
```

Comment:
What do you think having an `Encoder` type (interface), and then `Prover` is a subtype of `Encoder` (`is-a` relationship, which is struct embedding in go)? The "XAndY()" method looks like it could be decomposed in general.

Reply:
That sounds good to me as well. Then to be consistent, Decoder could also be a parallel concept and used under Framer?

Reply:
RSEncoder and Decoder are at the same level. "Framer" choose to include RSEncoder and Prover. The way that the current PR handles it is right to me. rs and kzgrs are at the same lelvel.